agreed/0009-MPID-Messaging/0009-MPID-Messaging.md

```diff
@@ -337,15 +327,15 @@ pub const MAX_BODY_SIZE: usize = ::structured_data::MAX_STRUCTURED_DATA_SIZE_IN_
 /// MpidMessage
 #[derive(Eq, PartialEq, PartialOrd, Ord, Clone, RustcDecodable, RustcEncodable)]
 pub struct MpidMessage {
-    header: ::MpidHeader,
-    recipient: ::NameType,
+    header: MpidHeader,
+    recipient: XorName,
     body: Vec<u8>,
     recipient_and_body_signature: ::sodiumoxide::crypto::sign::Signature,
 }
 
 impl MpidMessage {
-    pub fn new(header: ::MpidHeader,
-               recipient: ::NameType,
+    pub fn new(header: MpidHeader,
+               recipient: XorName,
                body: Vec<u8>,
                secret_key: &::sodiumoxide::crypto::sign::SecretKey)
                -> Result<MpidMessage, ::error::RoutingError> {
```

Comment:
I think there was the notion that the shared crate for vaults and clients would define these things like data error etc. So perhaps the errors would be best done there.